ingest/indexer/service/blockprocessor/pair_publisher.go

```diff
@@ -57,7 +59,13 @@ func (p PairPublisher) PublishPoolPairs(ctx sdk.Context, pools []poolmanagertype
 			spreadFactor := pool.GetSpreadFactor(ctx)
 			poolID := pool.GetId()
 
-			resultChan := make(chan error, len(denoms)*(len(denoms)-1)/2)
```

Comment:
The problem was this:

The formula is for the number of combinations of all `denoms` but notice that we later added logic to skip some denoms with `domain.ShouldFilterDenom(denomI)` so it ends up having to wait for more pairs than there actually are.

Reply:
I see! Thanks for the quick fix!